File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.swift
import Foundation

func solution(_ k:Int, _ tangerine:[Int]) -> Int {
    var dict: [Int: Int] = [:]
    
    tangerine.forEach { 
        if let _ = dict[$0] {
            dict[$0]! += 1
        } else {
            dict.updateValue(1, forKey: $0)
        }
    }
    
    let sortedKey = dict.keys.sorted {
        return dict[$0]! > dict[$1]!
    }

    var total = 0
    for i in 0..<sortedKey.count {
        let count = dict[sortedKey[i]]!
        
        if total + count >= k {
            return i + 1
        } else {
            total += count
        }
    }
    return 0
}

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(int k, vector<int> tangerine) {
    int answer = 0;
    sort(tangerine.begin(), tangerine.end());
    vector<int> count;
    
    for (int i = 0; i < tangerine.size(); ) {
        int j = i+1, cnt = 1;
        while (tangerine[i] == tangerine[j]) {
            cnt++;
            j++;
        }
        count.emplace_back(cnt);
        i = j;
    }
    sort(count.begin(), count.end());
    for (int i = count.size()-1; i >= 0; i--) {
        answer++;
        k -= count[i];
        if (k <= 0) break;
    }
    return answer;
}